### PR TITLE
Set maxNodeModuleJsDepth for inferred projects

### LINKED ISSUE #11116: inferred js projects do not get `maxNodeModuleJsDepth` set
``` cmd
mkdir test
cd test
npm init --y
npm install express3-handlebars
echo var exphbs = require('express3-handlebars'); > file.js
code .
```

now hover over `exphbs`, it shows `any`. 

create an empty `jsconfig.json` and reload the project, hover over it is a function.

the issue seems to be that we do not set the `maxNodeModuleJsDepth`for inferred projects.

### PR DESCRIPTION
**Pull request by zhengbli**
Fixes #11116

### CODE REVIEW

File: src/server/project.ts [hunk 1]
```diff
@@ -126,6 +126,13 @@ namespace ts.server {
                 this.compilerOptions.allowNonTsExtensions = true;
             }
 
+            if (this.projectKind === ProjectKind.Inferred) {
```
Comment (mhegazy) [inline src/server/project.ts]: should we check also that the project has .js files?

**Comment by mhegazy** [comment]
@zhengbli can you refresh this PR.

**Comment by zhengbli** [comment]
@mhegazy updated with a new implementation. Now the inferred project has a property called "isJsInferredProject" which serves as compilerOptions-independent flag so "js inferred project" can have different behavior than the normal inferred projects.

File: src/server/project.ts [hunk 2]
```diff
@@ -715,6 +712,26 @@ namespace ts.server {
             }
         })();
 
+        private _isJsInferredProject = false;
+        set isJsInferredProject(newValue: boolean) {
+            if (newValue && !this._isJsInferredProject) {
+                this.setCompilerOptions(this.getCompilerOptions());
```
Comment (vladima) [inline src/server/project.ts]: this looks odd

Comment (vladima) [inline src/server/project.ts]: also when before calling `setCompilerOptions` `this._isJsInferredProject` is not yet set so I'm not sure how control flow should enter `if` statement below
```
+            if (this._isJsInferredProject && typeof newOptions.maxNodeModuleJsDepth !== "number") {
 +                newOptions.maxNodeModuleJsDepth = 2;
 +            }
```

File: src/harness/unittests/tsserverProjectSystem.ts [hunk 3]
```diff
@@ -3051,4 +3051,29 @@ namespace ts.projectSystem {
             service.checkNumberOfProjects({ externalProjects: 1 });
         });
     });
+
+    describe("maxNodeModuleJsDepth for inferred projects", () => {
```
Comment (vladima) [inline src/harness/unittests/tsserverProjectSystem.ts]: can you have a test that actually verifies the value of `maxModuleDepth` (i.e. that we are not looking into folders that are located deeper)

File: src/server/project.ts [hunk 4]
```diff
@@ -715,6 +712,26 @@ namespace ts.server {
             }
         })();
 
+        private _isJsInferredProject = false;
+        set isJsInferredProject(newValue: boolean) {
```
Comment (vladima) [inline src/server/project.ts]: personal nit: I'm not a big fan of set-only accessors, can you turn it into method instead?

File: src/harness/unittests/tsserverProjectSystem.ts (hunk 3 above)
Comment (zhengbli) [inline src/harness/unittests/tsserverProjectSystem.ts]: Updated the test

**Comment by zhengbli** [comment]
Further comments? @mhegazy @vladima

**Pull request merged**